Use arducam-pivariety driver

platipus25 opened the pull request:
Signed-off-by: Ravago Jones <ravagojones@gmail.com>

platipus25 committed:
Use arducam-pivariety driver

We need to make the driver ignore the fact that it wants to be clocked at 24 Mhz
because the rock pi only wants to clock it at 6 MHz, and it still works.

Signed-off-by: Ravago Jones <ravagojones@gmail.com>

jkuszmaul commented:
Please add some comments to the commit mentioning the upstream source for these changes and what modifications you had to make.

platipus25 committed:
Copy arducam-pivariety from raspberrypi/linux

From branch rpi-6.0.7 at 6445b708fa4bcaaf3fe811ae8ac660b816300fc6

https://github.com/raspberrypi/linux/blob/rpi-6.0.y/drivers/media/i2c/arducam-pivariety.c

jkuszmaul commented on drivers/media/i2c/arducam-pivariety.c:
```diff
@@ -1357,9 +1357,8 @@ static int pivariety_probe(struct i2c_client *client,
 	pivariety->xclk_freq = clk_get_rate(pivariety->xclk);
 	if (pivariety->xclk_freq != 24000000) {
-		dev_err(dev, "xclk frequency not supported: %d Hz\n",
+		dev_warn(dev, "xclk frequency not supported: %d Hz\n",
```
If this warning is now gone, do we need to keep the diff?

platipus25 replied:
We need to keep it; this diff removes the error, and downgrades it to a warning.

AustinSchuh commented on arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi:
```diff
@@ -509,18 +509,18 @@
-                clock-names = "clk_cif_out";
-                //pinctrl-names = "rockchip,camera_default";
+                clock-names = "xclk";
+			          clock-frequency = <24000000>;
```
Can you be consistent with your indentation?

platipus25 replied:
Done
